src/muler/igrins.py

```diff
@@ -66,19 +70,29 @@ def __init__(self, *args, file=None, order=10, cached_hdus=None, **kwargs):
                 raise NameError("Cannot identify file as an IGRINS spectrum")
             grating_order = grating_order_offsets[band] + order
 
-            sn_file = file[:-13] + "sn.fits"
+            sn_file = file[19] + "sn.fits"
             if cached_hdus is not None:
                 hdus = cached_hdus[0]
                 sn_hdus = cached_hdus[1]
+                if wavefile is not None:
+                    wave_hdus = cached_hdus[2]
             else:
                 hdus = fits.open(str(file))
                 try:
                     sn_hdus = fits.open(sn_file)
                 except:
                     sn_hdus = None
+                if wavefile is not None:
+                    wave_hdus = fits.open(wavefile)
             hdr = hdus[0].header
-            lamb = hdus["WAVELENGTH"].data[order].astype(np.float64) * u.micron
-            flux = hdus["SPEC_DIVIDE_A0V"].data[order].astype(np.float64) * u.ct
+            if ".spec_a0v.fits" in file:
+                lamb = hdus["WAVELENGTH"].data[order].astype(np.float64) * u.micron
+                flux = hdus["SPEC_DIVIDE_A0V"].data[order].astype(np.float64) * u.ct
+            elif "spec.fits" in file and wavefile is not None:
+                lamb = wave_hdus[0].data[order].astype(np.float64) * 1e-3 * u.micron #Note .wave.fits and .wavesol_v1.fts files store their wavelenghts in nm so they need to be converted to microns
+                flux = hdus[0].data[order].astype(np.float64) * u.ct
+            else:
```

Comment:
I recommend adding a case here:

```python
elif ("spec_a0v.fits" in file) and (wavefile is not None):
   log.warn("You have passed in a wavefile and a spec_a0v format file, which has its own wavelength solution.  Ignoring the wavefile.")

elif ("spec.fits" in file) and (wavefile is None):
    raise Exception("wavefile must be specified when passing in spec.fits files, which do not come with an in-built wavelength solution.")
```

Reply:
I think the elif on line 98 will never trigger, since the first if on line 92 would always be satisfied first.  Maybe move the line 98 elif to an if at the start.

Reply:
Ah good catch.  Moved warning to first if.